docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
+---
+title: "How to convert an existing Gatsby blog to use MDX"
+date: 2019-11-10
+author: Thomas Wang
+excerpt: "It can be a big tricky to add MDX to an existing blog. This blog post covers how to install and configure MDX to Gatsby's blog starter"
+tags:
+  - blogs
+  - mdx
+  - markdown
+---
+
+## What is MDX?
+
+MDX is a file format that lets you add JSX to Markdown files via a `.mdx` file! It was started by and is maintained by Gatsby's [John Otander](https://github.com/johno) and the [open source community on Github](https://github.com/mdx-js/mdx).
+
+With MDX, you can import React components and declare them alongside regular markdown syntax. This is useful to display specific component examples alongside the actual code, complex interactive components like charts, and reusable components across multiple pages like sign-up forms.
+
+## Adding MDX to Gatsby Starter Blog
+
+It can be a big tricky to add MDX to an existing blog. The following 5 steps will cover how to install and configure MDX to [Gatsby's blog starter](https://github.com/gatsbyjs/gatsby-starter-blog/tree/master), which as of [today's version](https://github.com/gatsbyjs/gatsby-starter-blog/tree/8852c1e51651b902f19706ff2ca9f60dabc25709), does not have MDX pre-installed.
+
+You can also see the [full changes in PR #18929](https://github.com/gatsbyjs/gatsby/pull/18928/files) for an overview of the changes you have to make to get MDX working.
```

Comment:
Not sure if we want to link to a PR long term. Is there anything in there that isn't covered in this write up?

Reply:
i personally like these PR - because it shows on a living code how to change the code, sometimes textual description there can miss or hard understand some step, but the PR shows it detailed.

Reply:
I like it too, unless there's a better way to visualize all the changes in all files?